### PR TITLE
[TACHYON-1318] Add a findbug rule of finding unused variables

### PR DESCRIPTION
**Pull request by RongGu**
Add findbug rule for unread/unused private variables. The public/protected ones are allowed since they may be used with classes not seen as part of the analysis.

Also, this PR suppresses the found bugs with `@edu.umd.cs.findbugs.annotations.SuppressFBWarnings("URF_UNREAD_FIELD")`

### CODE REVIEW

File: servers/src/main/java/tachyon/worker/nio/NIODataServer.java [hunk 1]
```diff
@@ -61,6 +61,7 @@
   private Selector mSelector;
 
   // Instance of TachyonConf
+  @edu.umd.cs.findbugs.annotations.SuppressFBWarnings("URF_UNREAD_FIELD")
   private final TachyonConf mTachyonConf;
```
Comment (aaudiber) [inline servers/src/main/java/tachyon/worker/nio/NIODataServer.java]: We should be able to just remove this field

**Comment by aaudiber** [comment]
Thanks for the improvement to Findbugs coverage! I've left a few comments. Sorry some of them landed on a previous commit - was that because of a commit `--amend` and `push -f`?

File: servers/src/main/java/tachyon/worker/nio/NIODataServer.java (hunk 1 above)
Comment (RongGu) [inline servers/src/main/java/tachyon/worker/nio/NIODataServer.java]: Sure, I'll remove it.

**Comment by RongGu** [comment]
@aaudiber , thanks and I've addressed the comments.
Remove unused fields. Use `@SuppressFBWarnings` only. Use the 'compile' scope which is more understandable and add description for the dependency.

File: pom.xml [hunk 2]
```diff
@@ -283,6 +283,18 @@
         <version>${powermock.version}</version>
         <scope>test</scope>
       </dependency>
+
```
Comment (jsimsa) [inline pom.xml]: do we need this in the top-level package if the annotation is only used in the servers module?

Comment (swatDong) [inline pom.xml]: Currently only servers use. But this package can be used to manually handle any special findbug cases.
Both is OK for me.

Comment (RongGu) [inline pom.xml]: @aaudiber , any suggestions about putting to the top-level package?

Comment (aaudiber) [inline pom.xml]: I think top-level is fine as long as the scope keeps the dependency out of the jar.

**Comment by RongGu** [comment]
@aaudiber @jsimsa , thanks. I've changed the scope to `provided` and kept it still in top-level.

**Comment by jsimsa** [comment]
LGTM

**Comment by aaudiber** [comment]
LGTM

**Pull request merged**